quickwit-doc-mapping/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -0,0 +1,605 @@
+            FieldMappingType::Text(options) => {
+                vec![(
+                    field_path,
+                    FieldEntry::new_text(field_name, options.clone()),
+                )]
+            }
+            FieldMappingType::I64(options) => {
+                vec![(field_path, FieldEntry::new_i64(field_name, options.clone()))]
+            }
+            FieldMappingType::F64(options) => {
+                vec![(field_path, FieldEntry::new_f64(field_name, options.clone()))]
+            }
+            FieldMappingType::Date(options) => {
+                vec![(
+                    field_path,
+                    FieldEntry::new_date(field_name, options.clone()),
+                )]
+            }
+            FieldMappingType::Bytes(options) => {
+                vec![(
+                    field_path,
+                    FieldEntry::new_bytes(field_name, options.clone()),
+                )]
+            }
+            FieldMappingType::Object(field_mappings) => field_mappings
+                .iter()
+                .map(|mapping_entry| mapping_entry.field_entries())
+                .collect::<Result<Vec<_>, _>>()?
+                .into_iter()
+                .flatten()
+                .map(|(path, entry)| (path.with_parent(&self.name), entry))
+                .collect(),
+        };
+        Ok(results)
+    }
+
+    pub fn field_mappings(&self) -> Option<Vec<FieldMappingEntry>> {
+        match &self.mapping_type {
+            FieldMappingType::Object(entries) => Some(entries.clone()),
+            _ => None,
+        }
+    }
+
+    /// Returns tantivy field path and associated values to be added in the document.
+    // TODO: can be more efficient to pass a collector in argument (a kind of DocumentWriter)
+    // on which we add field values, thus we directly build the doucment instead of returning
+    // a Vec.
+    pub fn parse(
+        &self,
+        json_value: &JsonValue,
+    ) -> Result<Vec<(FieldPath, Value)>, DocParsingError> {
+        match &self.mapping_type {
+            FieldMappingType::Text(options) => self.parse_text(json_value, options),
+            FieldMappingType::I64(options) => self.parse_i64(&json_value, options),
+            FieldMappingType::F64(options) => self.parse_f64(&json_value, options),
+            FieldMappingType::Date(options) => self.parse_date(&json_value, options),
+            FieldMappingType::Bytes(options) => self.parse_bytes(&json_value, options),
+            FieldMappingType::Object(field_mappings) => {
+                self.parse_object(json_value, field_mappings)
+            }
+        }
+    }
+
+    pub fn parse_text(
+        &self,
+        json_value: &JsonValue,
+        _: &TextOptions,
+    ) -> Result<Vec<(FieldPath, Value)>, DocParsingError> {
+        let parsed_values = match json_value {
+            JsonValue::Array(_) => {
+                return Err(DocParsingError::MultiValuesNotSupported(self.name.clone()));
```

Comment:
we don't accept multiple values for text? why?

Reply:
Currently, I don't store the cardinality information, I linked fast field cardinality with the type cardinality but this is a bad choice. I need to store that info, and moreover concerning the fast field, we agreed that the cardinality will be always multivalued